src/loader/x86_64/elf/mod.rs

```diff
@@ -281,11 +284,18 @@ where
             .read_from(0, kernel_image, nhdr_sz)
             .map_err(|_| Error::ReadNoteHeader)?;
 
-        // If the note header found is not the desired one, keep reading until the end of the
-        // segment.
-        if nhdr.n_type == XEN_ELFNOTE_PHYS32_ENTRY {
-            break;
+        // If the note header found is not the desired one, keep reading until
+        // the end of the segment
```

Comment:
I would change this comment since you are adding checks other than the very simple one it had before. Perhaps something like:
``// Check if the note header's name and type match the ones specified by the PVH ABI.``